cvat-ui/src/components/common/cvat-table.tsx

```diff
@@ -0,0 +1,274 @@
+
+type Props = TableProps & {
+    onFilterDataSource?(data: TableProps['dataSource']): void;
+    onChangeColumnVisibility?(id: number, isHidden: boolean): void;
+    renderExtraActions?(): JSX.Element;
+    queryBuilderConfig?: Partial<Config>;
+    queryBuilderPredefinedQueries?: Record<string, string>;
+    queryBuilderCache?: {
+        key: string;
+        capacity: number;
+    };
+    csvExport?: {
+        filename: string;
+    };
+    tableTitle?: string;
+    searchDataIndex?: (string | string[])[];
+};
+
+/*
+    Enhanced table component
+    Supports:
+    * titles
+    * advanced filtration with json query
+    * overall search
+    * csv export
+    * show/hide columns
+    Current restrictions:
+    * queryBuilderConfig, queryBuilderPredefinedQueries, queryBuilderCache are supposed to be static
+*/
+function CVATTable(props: Props): JSX.Element {
+    const {
+        onChangeColumnVisibility,
+        onFilterDataSource,
+        renderExtraActions,
+        queryBuilderConfig,
+        queryBuilderCache,
+        queryBuilderPredefinedQueries,
+        searchDataIndex,
+        tableTitle,
+        dataSource,
+        csvExport,
+        columns,
+        ...rest
+    } = props;
+
+    const [FilteringComponent, setFilteringComponent] = useState<ReturnType<typeof ResourceFilterHOC> | null>(null);
+    const [filterValue, setFilterValue] = useState<string | null>(null);
+    const [searchPhrase, setSearchPhrase] = useState<string | null>(null);
+    const [visibility, setVisibility] = useState(defaultVisibility);
+    const [filteredDataSource, setFilteredDataSource] = useState<typeof dataSource>(dataSource);
+    const [modifiedColumns, setModifiedColumns] = useState<typeof columns>([]);
+
+    const downloadCSV = useCallback(() => {
+        if (csvExport && columns) {
+            // function to export as CSV properly handling commas, line breaks and double quotes in both header/fields
+            const header = columns.map((column) => ({
+                title: column.title ? `"${column.title}"` : undefined,
+                dataIndex: (column as any).dataIndex as string | string[],
+            })).filter((column) => !!column.title);
+
+            let csv = '';
+            if (dataSource) {
+                const rows = dataSource
+                    .map((dataItem) => header.map(({ dataIndex }) => {
+                        let value = null;
+                        if (Array.isArray(dataIndex)) {
+                            value = dataIndex.reduce((acc: unknown, path: string) => (
+                                typeof acc === 'object' && acc ? (acc as Record<string, unknown>)[path] : null
+                            ), dataItem);
+                        } else {
+                            value = dataItem[dataIndex];
+                        }
+
+                        if (typeof value === 'string') {
+                            return `"${value.replace(/"/g, '""')}"`;
+                        }
```

Comment:
Why do we need such manipulation with quotes? 
Also we can use `replaceAll` so we dont need to specify global flag.

Reply:
to support quotes in any fields.
See answer here: https://stackoverflow.com/questions/17808511/how-to-properly-escape-a-double-quote-in-csv

Reply:
also, in the past we had issues with replaceAll (its browser support low).
Maybe that is not a case anymore for us as in some cases we anyway use replaceAll now